hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/HoodieSyncClient.java

```diff
@@ -158,4 +171,23 @@ public List<PartitionEvent> getPartitionEvents(List<Partition> tablePartitions,
     }
     return events;
   }
+
+  /**
+   * Get Last commit's Metadata.
+   */
+  private static Option<HoodieCommitMetadata> getLatestCommitMetadata(HoodieTableMetaClient metaClient) {
```

Comment:
i think a more relevant logic we need for partition sync is: getCommitMetadataSinceLastSync(). it should return time-ordered commit metadata for extracting (partition, written|dropped) to be further produced partition events for sync

Reply:
Makes sense. Added a util method in TImelineUtils to get partitions dropped since give time